src/gp/src/GPMSA.C

```diff
@@ -894,25 +894,29 @@ GPMSAFactory<V, M>::setUpEmulator()
 
   const Map Wy_row_map(Wyrows, 0, comm);
 
-  m_observationErrorMatrix.reset
-    (new M(env, Wy_row_map, Wyrows));
-
   M& B = *m_BMatrix;
-  M& Wy = *m_observationErrorMatrix;
+
+  // Observation precision matrix
+  M Wy(env, Wy_row_map, Wyrows);
```

Comment:
I guess it's okay for Wy to be a local matrix now because we only use it indirectly via BT_Wy_Binv?